src/main/java/edu/kit/provideq/toolbox/ProblemController.java

```diff
@@ -1,62 +1,112 @@
+@RestController
 public abstract class ProblemController<ProblemFormatType, SolutionFormatType, SolverType extends ProblemSolver<ProblemFormatType, SolutionFormatType>> {
-  public abstract ProblemType getProblemType();
-  public abstract MetaSolver<SolverType> getMetaSolver();
+    @Lazy
+    @Autowired
+    private ProblemControllerProvider problemControllerProvider;
 
-  public SolutionHandle solve(SolveRequest<ProblemFormatType> request) {
-    Solution<SolutionFormatType> solution = SolutionManager.createSolution();
-    Problem<ProblemFormatType> problem = new Problem<>(request.requestContent, getProblemType());
+    @Autowired
+    private ApplicationContext context;
 
-    SolverType solver = getMetaSolver()
-            .getSolver(request.requestedSolverId)
-            .orElseGet(() -> getMetaSolver().findSolver(problem));
+    public abstract ProblemType getProblemType();
 
-    solution.setSolverName(solver.getName());
+    public abstract MetaSolver<SolverType> getMetaSolver();
 
-    long start = System.currentTimeMillis();
-    solver.solve(problem, solution);
-    long finish = System.currentTimeMillis();
+    public Solution<SolutionFormatType> solve(SolveRequest<ProblemFormatType> request) {
+        Solution<SolutionFormatType> solution = SolutionManager.createSolution();
+        Problem<ProblemFormatType> problem = new Problem<>(request.requestContent, getProblemType());
 
-    solution.setExecutionMilliseconds(finish - start);
+        SolverType solver = getMetaSolver()
+                .getSolver(request.requestedSolverId)
+                .orElseGet(() -> getMetaSolver().findSolver(problem));
 
-    return solution;
-  }
+        solution.setSolverName(solver.getName());
 
-  public SolutionHandle getSolution(long id) {
-    var solution = SolutionManager.getSolution(id);
-    if (solution == null) {
-      throw new ResponseStatusException(HttpStatus.NOT_FOUND,
-          String.format("Unable to find solution process with id %d", id));
+        SubRoutinePool subRoutinePool =
+                request.requestedSubSolveRequests == null
+                        ? context.getBean(SubRoutinePool.class)
+                        : context.getBean(SubRoutinePool.class,
+                        request.requestedSubSolveRequests
+                                .entrySet()
+                                .stream()
+                                .collect(Collectors.toMap(
+                                        Map.Entry::getKey,
+                                        e -> (Function<Object, Solution>) (content -> {
+                                            ProblemController problemController = problemControllerProvider.getProblemController(e.getKey());
+                                            SolveRequest solveRequest = e.getValue();
+                                            solveRequest.requestContent = content;
+                                            return (Solution) problemController.solve(solveRequest);
```

Comment:
We have a lot of raw generics all over the place. That's been with us since the initial draft of the system.
I didn't want to do a major refactor then without talking to anybody, but we can approach this issue in the refactoring step you mentioned in our meeting. For now I would keep it that way.